RDFSharp.UnitTests/RDFSelectQueryTest.cs

```diff
@@ -0,0 +1,193 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using RDFSharp.Query;
+using RDFSharp.Model;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+
+namespace RDFSharp.UnitTests
+{
+    [TestClass]
+    public class RDFSelectQueryTest
+    {
+        private static RDFGraph graph = TestRDFObjectProvider.graph;
+        private static string projectPath = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName;
+        private static string emptyQueryResultPath = Path.Combine(projectPath, "EmptyQueryResult.xml");
+        private static string emptyQueryTestResultPath = "EmptyQueryTestResult.xml";
+        private static string nonEmptyQueryResultPath = Path.Combine(projectPath, "NonEmptyQueryResult.xml");
+        private static string nonEmptyQueryTestResultPath = "NonEmptyQueryTestResult.xml";
+
+        private static string selectEverythingString = "SELECT *\n";
+        private static string foafPrefixString = "PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n\n";
+        private static RDFNamespace foafPrefix = RDFNamespaceRegister.GetByPrefix("foaf");
+        private static string selectString(List<string> variables)
+        {
+            var selectString = "SELECT";
+            foreach (var v in variables)
+            {
+                selectString += " ?" + v;
+            }
+            selectString += "\n";
+            return selectString;
+        }
+        private static string whereString(List<string> patternGroups = null)
+        {
+            var whereString = "WHERE {";
+            if (patternGroups != null)
+            {
+                foreach (var pg in patternGroups)
+                {
+                    whereString += "\n" + pg;
+                }
+            }
+            else
+                whereString += "\n";
+            whereString += "}\n";
+            return whereString;
+        }
+        private static string patternGroupString(List<string> patterns)
+        {
+            var patternGroupString = "  {";
+            foreach (var p in patterns)
+            {
+                patternGroupString += "\n    " + p + " .";
+            }
+            patternGroupString += "\n  }\n";
+            return patternGroupString;
+        }
+        private static string patternString(string subject, string predicate, string obj)
+        {
+            return "?" + subject + " " + predicate + " ?" + obj;
+        }
+
+
+        [TestMethod]
+        public void RDFSelectQuery_DefaultCtor_ReturnsGeneralSelectQuery()
+        {
+            //Arrange
+            var desiredResult = selectEverythingString + whereString();
```

Comment:
selectEverythingString helyett a "SELECT *\n" literál lehet, hogy átláthatóbb lenne